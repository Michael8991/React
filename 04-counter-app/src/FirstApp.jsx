import { Fragment } from "react"
import Porptypes from 'prop-types'

export const App = () => {
    return (
        <>
            <h1>{getResult()}</h1>
            <h1>Hola Mundo!</h1>
        </>
    )
}
