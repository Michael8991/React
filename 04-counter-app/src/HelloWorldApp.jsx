export const HelloWorldApp = () => {
    (
        <h1>Hello World App</h1>
    )
}