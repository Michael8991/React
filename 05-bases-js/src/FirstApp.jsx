import React from "react"
import PropTypes from "prop-types"

export const App = ({ title, subtitle, name }) => {
    return (
        <>
            <h1 data-testid="test-title">{title}</h1>
            <h2>{subtitle}</h2>
            <p>{name}</p>
        </>
    )
}

App.propType = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string,
    name: PropTypes.string,
}

App.defaultProps = {
    name: 'Michael Rodriguez',
    subtitle: 'Programando en React'
}
