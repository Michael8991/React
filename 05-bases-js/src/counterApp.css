body {
    height: 100vh;
    display: flex;
    align-items: center;
    justify-content: center;
    font-family: system-ui, -apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, Oxygen, Ubuntu, Cantarell, 'Open Sans', 'Helvetica Neue', sans-serif;
    background-color: rgb(0, 10, 43);
    color: white;
}

.titleCounterApp {
    width: 100%;
}

.counterContainer {
    width: 100%;
    display: flex;
    align-items: center;
    justify-content: center;
}

.counterButton {
    padding: 10px 15px;
    border-radius: 12px;
    border: none;
    background-color: rgb(0, 94, 131);
    color: white;
    box-shadow: 0px 0px 15px 5px rgba(0, 0, 0, 0.1);
    cursor: pointer;
    margin: 0 5px;
}

.counterButton:hover {
    scale: 1.10;
}

.totalCounter {
    display: flex;
    align-items: center;
    justify-content: center;
}