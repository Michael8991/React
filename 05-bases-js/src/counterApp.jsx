import React from 'react';
import { useState } from 'react'
import PropTypes from 'prop-types'

export const CounterApp = ({ value }) => {

    const [counter, setCounter] = useState(value);

    const sumar = () => setCounter(counter + 1)
    const restar = () => setCounter(counter - 1)
    const reset = () => setCounter(value)


    return (
        <>
            <h1 className='titleCounterApp'>CounterApp</h1>
            <p className='totalCounter'>Total: {counter}</p>
            <h2 className='counterContainer'>
                <button className='counterButton' onClick={sumar} type='button'>
                    +1
                </button>
                <button className='counterButton' onClick={restar} type='button'>
                    -1
                </button>
                <button className='counterButton' onClick={reset} type='button'>
                    Reset
                </button>
            </h2>
        </>
    )
}

CounterApp.propTypes = {
    // value: PropTypes.number.isRequired
};

