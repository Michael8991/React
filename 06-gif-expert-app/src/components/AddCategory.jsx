import PropTypes from 'prop-types';
import { useState } from 'react';

export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const onInputChanged = ({ target }) => {
        setInputValue(target.value)
    }

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim().length < 1) return;

        // setNewCategory(categories => [inputValue, ...categories])
        setInputValue("");
        onNewCategory(inputValue.trim())
    }

    return (
        <form onSubmit={(event) => onSubmit(event)} aria-label='form'>
            <input type="text" placeholder="Buscar..." value={inputValue} onChange={onInputChanged}></input>
        </form>
    )
}


AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}