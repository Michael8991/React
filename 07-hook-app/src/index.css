body {
    padding: 20px;
}

button {
    margin-right: 10px;
    /* background-color: #dadada; */
}