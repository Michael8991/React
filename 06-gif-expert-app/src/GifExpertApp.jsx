import { useState } from "react"
import { AddCategory } from "./components/AddCategory";
import { GifGrid } from "./components/GifGrid";

export const GifExpertApp = () => {

    const [categories, setCategories] = useState(['One Punch', 'Dragon Ball Z']);

    const onAddCategory = (onNewCategory) => {
        if (categories.includes(onNewCategory)) return;
        setCategories([onNewCategory, ...categories])
    }


    return (
        <>
            <h1>Gif Expert APP</h1>
            <AddCategory
                onNewCategory={(value) => onAddCategory(value)}
            />
            <ol>
                {
                    categories.map(category => (
                        <li key={category}>
                            <GifGrid
                                category={category} />
                        </li>
                    ))
                }
            </ol>

        </>
    )
}