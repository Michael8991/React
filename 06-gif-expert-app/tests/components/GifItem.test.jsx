import { render, screen } from "@testing-library/react"
import { GifExpertApp } from "../../src/GifExpertApp";
import { GifItem } from "../../src/components/GifItem";



describe('Pruebas en GifItem', () => {

    const title = 'Saitama';
    const url = 'https://media4.giphy.com/media/v1.Y2lkPWRjZWI2YWUyZ2o0M3hqaWgzZmZwYzFnb200OHpoOGM5aGE0Y2FidTc0OXl1ZHQ3NyZlcD12MV9naWZzX3NlYXJjaCZjdD1n/VXJWhaO7afRe/giphy.gif'

    test('Debe de hacer match con el snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />)
        expect(container).toMatchSnapshot();
    })

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
        render(<GifItem title={title} url={url} />)

        // expect(screen.getByRole('img').src).toBe(url)
        // expect(screen.getByRole('img').alt).toBe(title)

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url)
        expect(alt).toBe(title)
    })

    test('Debe de tener un titulo en el componente ', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy()
    })
})