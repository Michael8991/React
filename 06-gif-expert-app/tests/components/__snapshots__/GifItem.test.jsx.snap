// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en GifItem Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media4.giphy.com/media/v1.Y2lkPWRjZWI2YWUyZ2o0M3hqaWgzZmZwYzFnb200OHpoOGM5aGE0Y2FidTc0OXl1ZHQ3NyZlcD12MV9naWZzX3NlYXJjaCZjdD1n/VXJWhaO7afRe/giphy.gif"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
